app/services/mapping/goodsshipment/PreviousDocumentsBuilder.scala

```diff
@@ -33,28 +33,47 @@ object PreviousDocumentsBuilder {
   def build(implicit cacheMap: CacheMap): util.List[GoodsShipment.PreviousDocument] =
     cacheMap
       .getEntry[PreviousDocumentsData](Document.formId)
+      .filter(isDefined)
       .map(_.documents.map(createPreviousDocuments))
       .getOrElse(Seq.empty)
       .toList
       .asJava
 
+  private def isDefined(previousDocumentsData: PreviousDocumentsData): Boolean =
+    previousDocumentsData.documents.nonEmpty && previousDocumentsData.documents.forall(
+      doc =>
+        !doc.goodsItemIdentifier.getOrElse("").isEmpty ||
+          !Option(doc.documentReference).getOrElse("").isEmpty ||
+          !Option(doc.documentReference).getOrElse("").isEmpty ||
+          !Option(doc.documentCategory).getOrElse("").isEmpty
+    )
+
   private def createPreviousDocuments(document: Document): GoodsShipment.PreviousDocument = {
-    val categoryCode = new PreviousDocumentCategoryCodeType()
-    categoryCode.setValue(document.documentCategory)
+    val previousDocument = new GoodsShipment.PreviousDocument()
 
-    val id = new PreviousDocumentIdentificationIDType()
-    id.setValue(document.documentReference)
+    if (!Option(document.documentCategory).getOrElse("").isEmpty) {
```

Comment:
The same here about `!` and `isEmpty` and I don't understand why you wrapping something in Option and after that using GetOrElse

If `documentCategory` is String you can directly use isEmpty/nonEmpty on the string

Comment for lines 60 and 71 too.

Reply:
Good point!
Done! :)